Backend/src/main/java/ktsnvt/tim1/services/ReservationService.java

```diff
@@ -30,4 +53,74 @@ public ReservationDTO getReservation(Long id) throws EntityNotFoundException {
         return new ReservationDTO(reservationRepository.findById(id)
                 .orElseThrow(() -> new EntityNotFoundException("Reservation not found")));
     }
+
+    public ReservationDTO createReservation(NewReservationDTO newReservationDTO) throws EntityNotFoundException, EntityNotValidException, ImpossibleActionException {
+        Event event = eventRepository.findByIsActiveForReservationsTrueAndIsCancelledFalseAndById(newReservationDTO.getEventId())
+                .orElseThrow(() -> new EntityNotFoundException("Event not found"));
+        Date firstEventDay = event.getEventDays().stream().map(EventDay::getDate).min(Date::compareTo).get();
+        int numOfDaysToEvent = (int) TimeUnit.DAYS.convert(Math.abs(firstEventDay.getTime() - new Date().getTime()), TimeUnit.MILLISECONDS);
+        if (numOfDaysToEvent < 0) throw new EntityNotValidException("Event already started");
+        if (numOfDaysToEvent <= event.getReservationDeadlineDays())
+            throw new EntityNotValidException("Reservation deadline date passed");
+
+        HashSet<Ticket> tickets = new HashSet<Ticket>();
+        for (NewTicketDTO t : newReservationDTO.getTickets()) {
+            makeTicket(t, tickets);
+        }
+        Reservation reservation = new Reservation();
+        reservation.setEvent(event);
+        reservation.setTickets(tickets);
+        return new ReservationDTO(reservation);
+    }
+
+    private void makeTicket(NewTicketDTO ticketDTO, Set<Ticket> tickets) throws EntityNotFoundException, ImpossibleActionException {
```

Comment:
It would be better to split this method into 4 methods:
- seat reservation single day
- parterre reservation single day
- seat reservation all days
- parterre reservation all days

Reply:
Method split into 4 methods.